### PR TITLE
Custom buttons with borders rounded and gradients

### PR DESCRIPTION
**Pull request by joaquinbozzalla**
## Summary ##

**Trello card**: https://trello.com/c/KEUSqkOR

Custom buttons with border rounded and gradients. They are used in the login screen and in the book details.

### iPhone 11

simulator_screenshot_0FA5C534-FDCC-460A-A45D-0263B85A79BE

## Screencast ##

demo4

### CODE REVIEW

File: WBooks/Extensions/UIButton.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    
+    func makeCustomSolid() {
+        setTitleColor(.white, for: .normal)
+        contentEdgeInsets = UIEdgeInsets(top: 15.0, left: 0.0, bottom: 15.0, right: 0.0)
+
+        layoutIfNeeded()
+        let gradientLayer = CAGradientLayer()
+        gradientLayer.colors = [
+            UIColor(red: 0, green: 0.68, blue: 0.93, alpha: 1).cgColor,
+            UIColor(red: 0.22, green: 0.8, blue: 0.8, alpha: 1).cgColor
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIButton.swift]: move these colors to your UIColor extension

File: WBooks/Extensions/UIButton.swift [hunk 2]
```diff
@@ -8,6 +8,30 @@
...
+        gradientLayer.colors = UIColor.blueGradientColors
+        gradientLayer.startPoint = CGPoint(x: 0, y: 0.5)
+        gradientLayer.endPoint = CGPoint(x: 1, y: 0.5)
+        gradientLayer.cornerRadius = frame.height / 2
+        gradientLayer.frame = bounds
+        layer.insertSublayer(gradientLayer, at: 0)
+    }
+    
     func loadingIndicator(_ show: Bool) {
         let tag = 808404
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIButton.swift]: why do you need to set the tag?

Comment (joaquinbozzalla) [reply WBooks/Extensions/UIButton.swift]: I need it to identify the loading indicator in the `else` part to stop its animation. Maybe there is a better approach

File: WBooks/Extensions/UIButton.swift [hunk 3]
```diff
@@ -8,6 +8,30 @@
...
+        contentEdgeInsets = UIEdgeInsets(top: 15.0, left: 0.0, bottom: 15.0, right: 0.0)
+        
+        layer.borderWidth = 2.0
+        layer.borderColor = mainColor.cgColor
+        layer.cornerRadius = frame.height / 2
+    }
+    
+    func makeCustomSolid() {
+        setTitleColor(.white, for: .normal)
+        contentEdgeInsets = UIEdgeInsets(top: 15.0, left: 0.0, bottom: 15.0, right: 0.0)
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIButton.swift]: as I told you in slack. Set the button height in the interface builder and remove this contentEdgeInsets; also for showing the gradient inside the button you should add `clipToBunds = true` to your button.

with this you could set the style in `didSet` and remove `viewDidLayourSubviews`

**Review by carcos-wolox — APPROVED** [review]
with this, we finish the training 🎉 🎉   NICEE!!! 👏 👏

**Pull request merged**